### PR TITLE
CVAT-3D - Part1

### PR DESCRIPTION
**Pull request by manasars**
Changes Include:
1. Fix for Frame manipulation by mouse isn't activated by default in "select" mode
2. Fix for Shape selection works with a significant delay
3. Fix for cuboid draw issue 

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

### CODE REVIEW

**Comment by nmanovic** [comment]
@manasars , could you please merge the latest develop branch? It has some fixes to make the tests more stable. After that the PR should be green.

**Comment by manasars** [comment]
> @manasars , could you please merge the latest develop branch? It has some fixes to make the tests more stable. After that the PR should be green.

Hi @nmanovic : I merged the latest develop code , but now I see below error in tests:
Error: No such container: cvat

Looks like its the same behaviour as in other PRs.

File: cvat-canvas3d/src/typescript/canvas3dView.ts [hunk 1]
```diff
@@ -789,9 +816,21 @@ export class Canvas3dViewImpl implements Canvas3dView, Listener {
...
+            this.views.perspective.scene.background = new THREE.Color(canvasBackgroundColor);
+            this.views.side.scene.background = new THREE.Color(canvasBackgroundColor);
+            this.views.front.scene.background = new THREE.Color(canvasBackgroundColor);
+        } else if (reason === UpdateReasons.UPDATE_OPACITY) {
+            this.resetColor();
         } else if (reason === UpdateReasons.SHAPE_ACTIVATED) {
             const { clientID } = this.model.data.activeElement;
-            this.setupObjects();
+            if (clientID !== 'null') {
+                this.setupObjects(true);
```
Comment (bsekachev) [inline cvat-canvas3d/src/typescript/canvas3dView.ts:830]: Why do we need to pass the argument here (based on ``this.model.data.activeElement``)? In ``this.setupObjects`` we also get access to ``this.model.data.activeElement`` (right in the first line)

File: cvat-canvas3d/src/typescript/canvas3dView.ts [hunk 2]
```diff
@@ -750,12 +754,27 @@ export class Canvas3dViewImpl implements Canvas3dView, Listener {
...
-            this.clearSceneObjects();
+            if (!selected) {
+                this.clearSceneObjects();
+            }
+            this.resetColor();
             this.setHelperVisibility(false);
             for (let i = 0; i < this.model.data.objects.length; i++) {
                 const object = this.model.data.objects[i];
+                if (selected) {
+                    const target = this.views.perspective.scene.getObjectByName(clientID);
```
Comment (bsekachev) [inline cvat-canvas3d/src/typescript/canvas3dView.ts:768]: ``clientID`` is not changed in the loop, why do we get ``target`` identifier in the loop?

Comment (bsekachev) [reply cvat-canvas3d/src/typescript/canvas3dView.ts:768]: If target is const, than line: 

```this.views.perspective.scene.children[0].remove(target);`` tries to remove the same element a lot of times.
Please, correct me if I am wrong

File: cvat-canvas3d/src/typescript/canvas3dView.ts [hunk 3]
```diff
@@ -750,12 +754,27 @@ export class Canvas3dViewImpl implements Canvas3dView, Listener {
...
+                    if (`${object.clientID}` !== clientID) {
+                        continue;
+                    } else {
+                        this.views.perspective.scene.children[0].remove(target);
+                        this.views.side.scene.children[0].children = [];
+                        this.views.front.scene.children[0].children = [];
+                        this.views.top.scene.children[0].children = [];
```
Comment (bsekachev) [inline cvat-canvas3d/src/typescript/canvas3dView.ts:775]: Why do we need to clear side/front/top scenes a lot of times if only one object can be there at the same time?

File: cvat-canvas3d/src/typescript/canvas3dView.ts [hunk 4]
```diff
@@ -789,9 +816,21 @@ export class Canvas3dViewImpl implements Canvas3dView, Listener {
             loader.load(objectURL, this.addScene.bind(this));
             URL.revokeObjectURL(objectURL);
             this.dispatchEvent(new CustomEvent('canvas.setup'));
+        } else if (reason === UpdateReasons.UPDATE_CANVAS_BACKGROUND) {
+            const { canvasBackgroundColor } = this.model.data.shapeProperties;
+            this.views.top.scene.background = new THREE.Color(canvasBackgroundColor);
+            this.views.perspective.scene.background = new THREE.Color(canvasBackgroundColor);
+            this.views.side.scene.background = new THREE.Color(canvasBackgroundColor);
+            this.views.front.scene.background = new THREE.Color(canvasBackgroundColor);
```
Comment (bsekachev) [inline cvat-canvas3d/src/typescript/canvas3dView.ts:824]: Did you try to test this code?

**Comment by nmanovic** [comment]
@manasars , we are not going to access these changes. Too many new comments from our team.

**Pull request closed without merging**